pglookout/pglookout.py

```diff
@@ -86,82 +110,141 @@ def __init__(self, config_path):
         self.cluster_monitor.log.setLevel(self.log_level)
-        self.webserver = WebServer(self.config, self.cluster_state, self.cluster_monitor_check_queue)
+        self.webserver: WebServer = WebServer(self.config, self.cluster_state, self.cluster_monitor_check_queue)
+
+        if not self._is_initial_state_valid():
+            self.log.error("Initial state is invalid, exiting.")
+            sys.exit(1)
 
-        logutil.notify_systemd("READY=1")
+        notify_systemd("READY=1")
         self.log.info(
             "PGLookout initialized, local hostname: %r, own_db: %r, cwd: %r",
-            socket.gethostname(),
+            gethostname(),
             self.own_db,
-            os.getcwd(),
+            Path.cwd(),
         )
 
-    def quit(self, _signal=None, _frame=None):
+    def _is_initial_state_valid(self) -> bool:
+        """Check if the initial state of PgLookout is valid.
+
+        Note:
+            This method is only needed because we had to pick some default values before loading the config.
+            A better approach would be to load the config first, and then initialize PgLookout.
+            For that, :any:`load_config` should be split into two methods: one for loading the config, and one for
+            applying the config when doing hot reloads. The method loading the config should make minimal usage of
+            ``self`` and return a new ``Config`` object. The method applying the config should accept a ``Config``
+            object and apply it to ``self``. Meanwhile, ``__init__`` should use this ``Config`` object to initialize
+            itself, and it could be in different ways than during hot reloads (some internals don't depend on the config).
+
+        Note:
+            To developers: Any attribute that should be mandatory in the config should be checked here.
+        """
+        is_valid = True
+
+        if not self.config or not self.config_path.is_file():
+            self.log.error("Config is empty!")
+            is_valid = False
+
+        if not self.own_db:
+            self.log.error("`own_db` has not been set!")
+            is_valid = False
+
+        return is_valid
+
+    def quit(self, _signal: int | None = None, _frame: FrameType | None = None) -> None:
+        if self.cluster_monitor is None:
+            raise RuntimeError("Cluster monitor is not initialized!")
+
         self.log.warning("Quitting, signal: %r, frame: %r", _signal, _frame)
         self.cluster_monitor.running = False
         self.running = False
         self.webserver.close()
 
-    def load_config(self, _signal=None, _frame=None):
+    def load_config_from_signal(self, _signal: int, _frame: FrameType | None = None) -> None:
         self.log.debug(
             "Loading JSON config from: %r, signal: %r, frame: %r",
             self.config_path,
             _signal,
             _frame,
         )
+        self.load_config()
```

Comment:
This new method is the only one expected to have access to signal and frame. This type of info should not reach the `load_config`. The behaviour is left unchanged.

However, now, that means that `load_config` can be called as a normal method.